server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -698,53 +698,55 @@ private synchronized IndexService createIndexService(
+                idxSettings.getNumberOfShards(),
+                idxSettings.getNumberOfReplicas(),
+                indexCreationContext
+            );
 
-        final IndexModule indexModule = new IndexModule(
-            idxSettings,
-            analysisRegistry,
-            getEngineFactory(idxSettings),
-            directoryFactories,
-            () -> allowExpensiveQueries,
-            indexNameExpressionResolver,
-            recoveryStateFactories
-        );
-        for (IndexingOperationListener operationListener : indexingOperationListeners) {
-            indexModule.addIndexOperationListener(operationListener);
-        }
-        pluginsService.forEach(p -> p.onIndexModule(indexModule));
-        for (IndexEventListener listener : builtInListeners) {
-            indexModule.addIndexEventListener(listener);
+            final IndexModule indexModule = new IndexModule(
+                idxSettings,
+                analysisRegistry,
+                getEngineFactory(idxSettings),
+                directoryFactories,
+                () -> allowExpensiveQueries,
+                indexNameExpressionResolver,
+                recoveryStateFactories
+            );
+            for (IndexingOperationListener operationListener : indexingOperationListeners) {
+                indexModule.addIndexOperationListener(operationListener);
+            }
+            pluginsService.forEach(p -> p.onIndexModule(indexModule));
+            for (IndexEventListener listener : builtInListeners) {
+                indexModule.addIndexEventListener(listener);
+            }
+            return indexModule.newIndexService(
+                indexCreationContext,
+                nodeEnv,
+                parserConfig,
+                this,
+                circuitBreakerService,
+                bigArrays,
+                threadPool,
+                scriptService,
+                clusterService,
+                client,
+                indicesQueryCache,
+                mapperRegistry,
+                indicesFieldDataCache,
+                namedWriteableRegistry,
+                idFieldMappers.apply(idxSettings.getMode()),
+                valuesSourceRegistry,
+                indexFoldersDeletionListeners,
+                snapshotCommitSuppliers
+            );
         }
-        return indexModule.newIndexService(
-            indexCreationContext,
-            nodeEnv,
-            parserConfig,
-            this,
-            circuitBreakerService,
-            bigArrays,
-            threadPool,
-            scriptService,
-            clusterService,
-            client,
-            indicesQueryCache,
-            mapperRegistry,
-            indicesFieldDataCache,
-            namedWriteableRegistry,
-            idFieldMappers.apply(idxSettings.getMode()),
-            valuesSourceRegistry,
-            indexFoldersDeletionListeners,
-            snapshotCommitSuppliers
-        );
     }
```

Comment:
Do we need to suppress warnings from this whole method, or could we reduce the scope?

Reply:
Highlighted the places that emit the warning. I guess, it could be reduced

Reply:
Thanks for the extra detail. That's pretty much the whole method, let's leave it as it is.

Reply:
Oh, wait, I see, this is a new method that just creates the `IndexModule`. Great, thanks for isolating it like this.